ui/app/AppLayouts/Profile/panels/TokenSettingsModalContent.qml

```diff
@@ -0,0 +1,151 @@
+import QtQuick 2.13
+import QtQuick.Controls 2.13
+import QtQuick.Layouts 1.13
+
+import utils 1.0
+
+import StatusQ.Controls 0.1
+import StatusQ.Components 0.1
+
+import shared.panels 1.0
+import shared.popups 1.0
+import shared.controls 1.0
+
+Item {
+    id: modalBody
+
+    property var defaultTokenList
+    property var customTokenList
+    signal toggleVisibleClicked(int chainId, string symbol)
+    signal removeCustomTokenTriggered(int chainId, string address)
+    signal showTokenDetailsTriggered(int chainId, string address, string name, string symbol, string decimals)
+
+    Component {
+        id: tokenComponent
+
+        StatusListItem {
+            id: assetSymbol
+            title: symbol
+            subTitle: name || ""
+            image.source: Style.png("tokens/" + (hasIcon ? symbol : "DEFAULT-TOKEN@3x"))
+            image.height: 36
+            components: [StatusCheckBox {
+                id: assetCheck
+                checked: model.isVisible
+                onClicked: toggleVisibleClicked(chainId, symbol)
+            }]
+            visible: symbol && (searchBox.text == "" || name.toLowerCase().includes(searchBox.text.toLowerCase()) || symbol.toLowerCase().includes(searchBox.text.toLowerCase()))
+            MouseArea {
+                acceptedButtons: Qt.LeftButton | Qt.RightButton
+                cursorShape: Qt.PointingHandCursor
+                anchors.fill: parent
+                onClicked: function (event) {
+                    if (event.button === Qt.RightButton) {
+                        return contextMenu.popup(mouseX, mouseY)
+                    }
+                    assetCheck.checked = !assetCheck.checked
+                    toggleVisibleClicked(chainId, symbol)
+                }
+                
+                // TODO: replace with StatusPopupMenu
```

Comment:
this to StatusPopupMenu? I missed this before

Reply:
done